src/validators/is-allowance-program.validator.ts

```diff
@@ -21,7 +21,7 @@ export class IsAllowanceProgramValidator
       found = await this.entityManager.findOneBy(ProgramCode, {
         programCode: value.toUpperCase(),
         allowanceUIFilter: 1,
-        tradingEndDate: IsNull(),
+        tradingEndDate: null,
```

Comment:
The version of TypeORM we're using treats `null` the same as `undefined` in queries, i.e. it strips the condition from the WHERE clause.  In this case, we _do_ want to ensure that `tradingEndDate` is null when the `isActiveOnly` flag is set.

However, that flag is set to `false` in all invocations of the `IsAllowanceProgram` pipe, so this block shouldn't ever be entered. It is being entered because of a bug in extracting the constraints above: each array item is an object, so object destructuring should be used to extract the relevant property in line 18 above:
```typescript
const { isActiveOnly } = args.constraints[0];
```